packages/itwin/grouping-mapping-widget/src/components/Mappings/MappingList.tsx

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { Mapping } from "@itwin/insights-client";
+import React, { useEffect } from "react";
+import type { BeEvent } from "@itwin/core-bentley";
+import { MappingUIActionGroup } from "./MappingViewActionGroup";
+import { useState } from "react";
+import { useGroupingMappingApiConfig } from "../context/GroupingApiConfigContext";
+import { Anchor, ListItem, Text } from "@itwin/itwinui-react";
+import { ExtractionStates } from "./Extraction/ExtractionStatus";
+import { ExtractionStatus } from "./Extraction/ExtractionStatus";
+import { useFetchExtractionStates } from "./hooks/useFetchExtractionStates";
+import "./MappingList.scss";
+
+export interface MappingListProps {
+  selected: boolean;
+  onSelectionChange: (mapping: Mapping) => void;
+  mapping: Mapping;
+  jobStartEvent: BeEvent<(mappingId: string) => void>;
+  onClickMappingTitle?: (mapping: Mapping) => void;
+  onClickMappingModify?: (mapping: Mapping) => void;
+  onRefreshMappings: () => Promise<void>;
+  onToggleExtraction: (mapping: Mapping) => Promise<void>;
+  setShowDeleteModal: (mapping?: Mapping) => void;
+}
+
+export const MappingList = (props: MappingListProps) => {
+  const [extractionState, setExtractionState] = useState<ExtractionStates | undefined>(ExtractionStates.None);
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const { extractionStateData, isJobStarted, isJobDone, setIsJobDone, setIsJobStarted } = useFetchExtractionStates(groupingMappingApiConfig);
+
+  const onClickTile = () => {
+    props.onSelectionChange(props.mapping);
+  };
+
+  useEffect(() => {
+    const listener = (startedMappingId: string) => {
+      if (startedMappingId === props.mapping.id) {
+        setExtractionState(ExtractionStates.Starting);
+        setIsJobStarted(true);
+      }
+    };
+    props.jobStartEvent.addListener(listener);
+
+    return () => {
+      props.jobStartEvent.removeListener(listener);
+    };
+  }, [props.jobStartEvent, props.mapping.id, setIsJobStarted]);
+
+  useEffect(() => {
+    if (isJobStarted === true || isJobDone === true) {
+      setExtractionState(extractionStateData);
+    }
+  }, [extractionStateData, isJobDone, isJobStarted]);
+
+  return (
+    <ListItem actionable
+      className="gmw-list-item-container"
+      active={props.selected}
+      key={props.mapping.id}
+      onClick={onClickTile}
+      title={props.mapping.mappingName}>
+      <ListItem.Content>
```

Comment:
The contents of `ListItem.Content` are hard to read as they currently are. Could you split the inner components into multiple lines? Something similar like you have for `ListItem` - the component starts at one line and each of the properties get their own line. Then the ternary operator could also be split into multiple lines.

Reply:
I have made this change.